addons/hw_drivers/controllers/driver.py

```diff
@@ -40,8 +40,7 @@ def action(self, session_id, device_identifier, data):
                     _logger.info("Ignored request from %s as iot_idempotent_id %s already received from session %s",
                                  session_id, iot_idempotent_id, idempotent_session)
                     return False
-            iot_device.action(data)
-            return True
+                return iot_device.action(data) if not None else True
```

Comment:
what is the purpose of this part ?

I guess that you try to don't always return `True` if the function return something. But I am a bit confused.

I am also a bit afraid that it will have side effects:
let's assume that `iot_device.action(data)` return `None`, in this case, with your code it will also return `None` which I doubt it is intended. In general we prefer to use `False` if something fail.
Note that in python you can use `or` and `and` to easily alter the value of boolean

Reply:
The intent is indeed to keep the default behaviour: return True (as before this commit) unless `iot_device.action(data)` returned a value.
From my understanding, there are 3 cases:
- `iot_device.action(data)` returns True
- `iot_device.action(data)` returns False
- `iot_device.action(data)` returns None => True will be returned instead

Excuse the newbie Python from my end, I think this will make more sense: 
```
result = iot_device.action(data)
return result if result is not None else True
```

The reason I want to handle the case where the value returned is `None` as no actions returns a value except for actions related to printing.